tidy3d/components/data/sim_data.py

```diff
@@ -953,13 +956,18 @@ def source_spectrum_fn(freqs):
 
     def make_adjoint_sim(
         self, data_vjp_paths: set[tuple], adjoint_monitors: list[Monitor]
-    ) -> Simulation:
+    ) -> tuple[Simulation, float]:
```

Comment:
Wondering if it would make sense to introduce some info object here similar to the `DerivativeInfo` that we have already instead of carrying around (a potentially increasing number of) outputs.

Reply:
After reading through the whole thing, maybe it would make sense to return, instead of a simulation, some `AdjointData` type similar to `SimulationData`, that containts the simulation as well as other info we might need? Not sure if it's a good idea but maybe worth considering.

Reply:
yea that might be a good idea (similar to `DerivativeInfo`?). I could do that to clean things up.

Reply:
alright, I made am `AdjointSourceInfo` in `sim_data.py` to store everything. seems cleaner and I can extend it easily later.

```py
class AdjointSourceInfo(Tidy3dBaseModel):
    """Stores information about the adjoint sources to pass to autograd pipeline."""

    sources: tuple[Source, ...] = pd.Field(
        ...,
        title="Adjoint Sources",
        description="Set of processed sources to include in the adjoint simulation."
    )

    post_norm: Union[float, DataArray] = pd.Field(
        ...,
        title="Post Normalization Values",
        description="Factor to multiply the adjoint fields by after running "
        "given the adjoint source pipeline used."
    )

    normalize_sim: bool = pd.Field(
        ...,
        title="Normalize Adjoint Simulation",
        description="Whether the adjoint simulation needs to be normalized "
        "given the adjoint source pipeline used."
    )

```